src/shared/components/close-button/CloseButton.tsx

```diff
@@ -35,4 +35,51 @@ const CloseButton = (props: Props) => {
   );
 };
 
+/**
+ * Design mock-ups are showing a common pattern of a close button
+ * accompanied by a short label either to the left or to the right of the button.
+ * Both the label and the button are clickable.
+ * This component captures this simple arrangement.
+ * For anything more complicated, please use constituent components separately.
+ */
+export const CloseButtonWithLabel = (
+  props: Props & {
+    label?: string;
+    labelPosition?: 'left' | 'right';
+  }
+) => {
+  const {
+    label = 'Close',
+    labelPosition = 'left',
+    onClick,
+    className: classNameFromProps
+  } = props;
+
+  const componentClasses = classNames(
+    styles.closeButtonWithLabel,
+    classNameFromProps
+  );
+
+  const closeIcon = <CloseIcon className={styles.icon} />;
+
+  const buttonContent =
+    labelPosition === 'left' ? (
+      <>
+        <span>{label}</span>
+        {closeIcon}
+      </>
+    ) : (
+      <>
+        {closeIcon}
+        <span>{label}</span>
+      </>
+    );
```

Comment:
If a label is not passed as props then will there be an empty `span` tag?

Reply:
No; on line 52 the label is assigned a default value, "Close".